logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageRepository.kt

```diff
@@ -334,8 +343,9 @@ class MessageDataSource(
                 else -> networkFailure
             }
             Either.Left(failure)
-        }, {
-            Either.Right(it.time)
+        }, { response: QualifiedSendMessageResponse ->
+            // not sure about this forced cast, but it seems should be the case here to receive a QualifiedSendMessageResponse.MessageSent
+            Either.Right(SendMessagePartialFailureMapperImpl.fromDTO(response as QualifiedSendMessageResponse.MessageSent))
```

Comment:
Couldn't the two folds with `sendMessageFailureMapper` and `SendMessagePartialFailureMapperImpl` go away and be replaced with:

```
when(response) {
  is QualifiedSendMessageResponse.MessageSent -> ...
  is QualifiedSendMessageResponse.MissingDevicesResponse -> ..
}

Reply:
Delegated this `when` and mapping down to the mapper =)

Reply:
Mm I didn't quite understand the this code, also not so cool to have case which never will happen. 

I think we should consider mapping the 412 to be a success so that both response cases can be handled in a single `when` statement.

But should probably happen in a separate PR.